### PR TITLE
[Dynamo] Add a dummy profiler to avoid activating real profiler

### PR DESCRIPTION
**Pull request by yanboliang**
See context at https://github.com/pytorch/torchdynamo/issues/1721#issuecomment-1312396059

cc @mlazos @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: torch/autograd/profiler.py [hunk 1]
```diff
@@ -441,6 +441,20 @@ def createFunctionEventForMemoryEvents(evt):
         return function_events
 
 
+class dummy(object):
```
Comment (jansel) [inline torch/autograd/profiler.py:444]: This class looks the same as `contextlib.nullcontext()`.

**Review by jansel — APPROVED** [review]

**Comment by yanboliang** [comment]
Thank you for reminding me of ```contextlib.nullcontext```, I decided to switch to that. It'll be easily used for other context managers we don't support yet, e.g, ```with torch.backends.cudnn.flags```(https://github.com/pytorch/torchdynamo/issues/1784).

File: torch/_dynamo/variables/user_defined.py [hunk 2]
```diff
@@ -77,8 +78,11 @@ def call_function(
 
         options = VariableTracker.propagate(self, args, kwargs.values())
 
-        if self.value is torch.autograd.profiler.profile:
-            return ProfilerContextWrapperVariable()
+        if self.value in (
+            contextlib.nullcontext,
+            torch.autograd.profiler.profile,
```
Comment (yuguo68) [inline torch/_dynamo/variables/user_defined.py:83]: do we still need this line?

Comment (yanboliang) [reply torch/_dynamo/variables/user_defined.py:83]: Yes, it will be used to rewrite ```with torch.autograd.profiler.profile()``` in users' code.

**Pull request closed without merging**